draft-enghardt-panrg-path-properties.md

```diff
@@ -96,42 +96,41 @@ in Path-Aware Networking {{I-D.irtf-panrg-questions}}, which is a product of the
 
 Path element:
 : Either a node or a link.
 
 Path:
-: A sequence of adjacent path elements, alternating between nodes and links, starting and ending with a host. A path can be viewed as an abstraction on a specific layer, omitting lower layer path elements. For example, a router implementing IPv6 may be a path element on a path when considering the network layer. If this router does not implement transport layer functionality, it is hidden when a higher layer, such as the transport or application layer, is considered. In the case of multicast or broadcast, a single packet may be sent over multiple paths at once -- one path for each combination of sending and receiving host.
+: A sequence of adjacent path elements over which a packet can be transmitted, starting and ending with a node. Note that paths are time-dependent, i.e., the sequence of path elements over which packets are sent between two nodes may change frequently. Furthermore, an entity may have only partial visibility of the path elements that comprise a path.
 
 Subpath:
-: Given a path, a subpath is a sequence of adjacent path elements of this path, starting and ending with a node.
+: Given a path, a subpath is a sequence of adjacent path elements of this path.
 
 Flow:
-: One or multiple packets which are traversing the same subpath or path. For example, a flow can consist of all packets sent within a TCP session with the same five-tuple between two hosts, or it can consist of all packets sent on the same physical link.
+: An entity made of packets to which the traits of a path or set of subpaths may be applied in a functional sense. For example, a flow can consist of all packets sent within a TCP session with the same five-tuple between two hosts, or it can consist of all packets sent on the same physical link.
 
 Property:
-: A trait of one or a sequence of path elements, or a trait of a flow with respect to one or a sequence of path elements. An example of a link property is the maximum data rate that can be sent over the link. An example of a node property is the administrative domain that the node belongs to. An example of a property of a flow with respect to a subpath is the aggregated one-way delay of the flow being sent from one node to another node over a subpath.
-A property is thus described by a tuple containing the sequence of path elements, the flow or an empty set if no packets are relevant for the property, the name of the property (e.g., maximum data rate), and the value of the property (e.g., 1Gbps).
+: A trait of one or a sequence of path elements, or a trait of a flow with respect to one or multiple subpaths. An example of a link property is the maximum data rate that can be sent over the link. An example of a node property is the administrative domain that the node belongs to. An example of a property of a flow with respect to a subpath is the aggregated one-way delay of the flow being sent from one node to another node over this subpath.
```

Comment:
Did you change the Property definition (on one or multiple subpaths) to allow properties on links between multiple nodes (e.g., wireless links?) Or is there some other motivation?

Reply:
Sorry I just now saw your comment on this pull request...
I'm thinking that for properties between endpoints, we might need an abstract link analogous to the abstract node. Then we can model the (sub-)paths between endpoints as two nodes/hosts connected by the abstract link (e.g., 5-tuple).
This approach which associates a property with multiple (sub-)paths could work as well but might have the issue that it is not clear when the property held for which of the (sub-)paths.

Reply:
Right, I see. You're right, having multiple subpaths in there might get too confusing, so I'll change it back. A property should apply to one or a sequence of path elements (and maybe a flow), so only one subpath.
I like your idea of an abstract link: For cases in which this subpath changes frequently, either we care and have insight into these path elements, or we do not care and just use an abstract link.

Reply:
Agree, let's keep it like this for now and see what the Research Group thinks.